### DISCUSSION TITLE
MicroPython v1.19 on 2022-06-23; linux [GCC 9.4.0] on Raspberry pi 4b Code #2

### DISCUSSION THREAD

**Discussion by FlydGitHub** [micropython/micropython#11225] [Other]
import time
from machine import Pin, PWM

# Define the maximum achievable PWM frequency in Hz
pwm_freq = 100000

# Define the duty cycle (0-1023)
duty_cycle = 512

# Define the GPIO pin for the PWM output in BOARD mode
pwm_pin = 40

# Configure the PWM output with the specified frequency and duty cycle
pwm = PWM(Pin(pwm_pin))
pwm.freq(pwm_freq)
pwm.duty(duty_cycle)

# Wait indefinitely
try:
    while True:
        time.sleep(1)
except KeyboardInterrupt:
    pass

**Comment by peterhinch** [comment]
The Unix port of MicroPython doesn't support pin I/O. If you want to run this kind of code you could use a Pico or Pico W.

**Comment by FlydGitHub** [comment]
Thanks Peter,

My plan was to use the RP 4B OS as a development environment for the Pico.

**Reply by peterhinch** [reply]
That should be fine, but I/O code will have to run on the Pico. The Unix build typically runs on PC's that don't support pin I/O.
